weather_app/weather_notifier.py

```diff
@@ -0,0 +1,43 @@
+import requests
+import os
+from dotenv import load_dotenv
+from plyer import notification
+
+# Load API Key
+load_dotenv()
+API_KEY = os.getenv("API_KEY")
+
+# Set city and API URL
+CITY = "Osaka"
+URL = f"http://api.openweathermap.org/data/2.5/weather?q={CITY}&appid={API_KEY}&units=metric"
```

Comment:
This looks good, except for two things
1. If this API happens to have more query params, you would need to add them to this string. If you add like 5, then it becomes this long string that is not easily readable. What you can do instead, is pass a dictionary to the post payload and use it as params. Requests will automatically convert that dictionary into the URL params you are currently using. https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls
2. It is okay for this small example, but usually, these third-party URLs are defined in the .env files as well. Reasoning is, you would want a different one per environment (dev vs prod, for example)